tests/unit/amo/components/TestCollectionManager.js

```diff
@@ -738,6 +738,27 @@ describe(__filename, () => {
     expect(root.find(Notice)).toHaveLength(0);
   });
 
+  it('calls clearTimeout when unmounting and timeout is set', () => {
+    const clearTimeoutSpy = sinon.spy();
+    const root = render({ clearTimeout: clearTimeoutSpy });
+
+    // Simulates the setTimeout behavior in which timeout is set.
+    root.instance().timeout = 50;
+
+    root.unmount();
+
+    expect(clearTimeoutSpy.callCount).toEqual(1);
+  });
+
+  it('does not call clearTimeout when unmounting and there is no timeout set', () => {
+    const clearTimeoutSpy = sinon.spy();
+    const root = render({ clearTimeout: clearTimeoutSpy });
+
+    root.unmount();
+
+    expect(clearTimeoutSpy.callCount).toEqual(0);
```

Comment:
We should use `sinon` for this kind of assertions, because `sinon` gives us a better output:

```diff
diff --git a/tests/unit/amo/components/TestCollectionManager.js b/tests/unit/amo/components/TestCollectionManager.js
index f23bd5b92..062f2882f 100644
--- a/tests/unit/amo/components/TestCollectionManager.js
+++ b/tests/unit/amo/components/TestCollectionManager.js
@@ -756,7 +756,7 @@ describe(__filename, () => {
 
     root.unmount();
 
-    expect(clearTimeoutSpy.callCount).toEqual(0);
+    sinon.assert.notCalled(clearTimeoutSpy);
   });
 
   it('removes the notification after a new add-on has been selected', () => {

```

Reply:
ahh..ok.. that makes sense